lib/deploy/functions/release.js

```diff
@@ -254,9 +189,9 @@ module.exports = function(context, options, payload) {
     var functionFilterGroups = _getFilterGroups();
     var releaseNames = _getReleaseNames(uploadedNames, existingNames, functionFilterGroups);
 
-    logFilters(existingNames, releaseNames, functionFilterGroups);
+    _logFilters(existingNames, releaseNames, functionFilterGroups);
 
-    var addOps = _.chain(uploadedNames)
+    var createOps = _.chain(uploadedNames)
       .difference(existingNames)
       .intersection(releaseNames)
       .map(function(functionName) {
```

Comment:
IIUC, you are no longer using the return value of this function but instead adding to an external `deployments` array. Two things:

1. Let's make this a forEach
2. Below it looks like you're using a `Promise.all()` that awaits this. But since you're not returning, there is no async resolution that will happen for this so it's going to immediately resolve.